libs/ml/include/ml/ops/ops.hpp

```diff
@@ -149,6 +150,57 @@ class Ops
     // Empty deafult implementation for non-trainable Ops.
   }
 
+  /**
+   * @brief ChargeForward
+   * @return estimated charge amount, necessary for performing a forward pass on data of given
+   * shapes.
+   */
+  virtual MLChargeAmount ChargeForward()
+  {
+    // TODO(VH): make a pure virtual call;
+    std::cout << "Error: call to unexisting ChargeForward() implementation! returned 0."
+              << std::endl;
+    return 0;
+  }
+
+  /**
+   * @brief ChargeBackward
+   * @return estimated charge amount, necessary for performing a backward pass on data of given
+   * shapes.
+   */
+  virtual MLChargeAmount ChargeBackward()
+  {
+    // TODO(VH): make a pure virtual call;
+    std::cout << "Error: call to unexisting ChargeBackward() implementation! returned 0."
+              << std::endl;
+    return 0;
+  }
+
+  /**
+   * @brief TotalElementsIn calculated a sum of total elements in all given tensors
+   * @param shapes - vector of tensor shapes
+   */
+  static fetch::math::SizeType TotalElementsIn(std::vector<fetch::math::SizeVector> const &shapes)
+  {
+    if (shapes.empty())
+    {
+      return 0;
+    }
+    fetch::math::SizeType total_elements = 1;
+    for (auto const &shape : shapes)
+    {
+      for (auto const &dimension : shape)
+      {
+        if (dimension == 0)
+        {
+          std::runtime_error("A dimension of size 0 found in tensor shape!");
```

Comment:
i guess this should be an ML exception of some kind

Reply:
I guess fetch::math::exceptions::WrongShape would fit here

Reply:
(this exception is available because we include a math::Tensor anyway)